plonky2/src/fri/batch_oracle.rs

```diff
@@ -0,0 +1,115 @@
+#[cfg(not(feature = "std"))]
+use alloc::vec;
+#[cfg(not(feature = "std"))]
+use alloc::vec::Vec;
+
+use itertools::Itertools;
+use plonky2_field::extension::Extendable;
+use plonky2_field::fft::FftRootTable;
+use plonky2_field::polynomial::{PolynomialCoeffs, PolynomialValues};
+use plonky2_maybe_rayon::*;
+use plonky2_util::{log2_strict, reverse_index_bits_in_place};
+
+use crate::fri::oracle::PolynomialBatch;
+use crate::hash::field_merkle_tree::FieldMerkleTree;
+use crate::hash::hash_types::RichField;
+use crate::plonk::config::GenericConfig;
+use crate::timed;
+use crate::util::timing::TimingTree;
+use crate::util::transpose;
+
+/// Represents a batch FRI oracle, i.e. a batch of polynomials with different degrees which have
+/// been Merklized in a Field Merkle Tree.
```

Comment:
Perhaps 'Merkle-ized' or 'Merkleized' (or 'Merkleised' if you want to speak the King's English.)

Merkle is the name of the inventor.